token/client/src/token.rs

```diff
@@ -1148,6 +1150,95 @@ where
         .await
     }
 
+    /// Transfer tokens confidentially with fee
+    #[allow(clippy::too_many_arguments)]
+    pub async fn confidential_transfer_transfer_with_fee<S2: Signer>(
+        &self,
+        source_token_account: &Pubkey,
+        destination_token_account: &Pubkey,
+        source_token_authority: &S2,
+        amount: u64,
+        source_available_balance: u64,
+        source_elgamal_keypair: &ElGamalKeypair,
+        new_source_decryptable_available_balance: AeCiphertext,
+        epoch_info: &EpochInfo,
+    ) -> TokenResult<T::Output> {
+        let source_state = self.get_account_info(source_token_account).await.unwrap();
+        let source_extension =
+            source_state.get_extension::<confidential_transfer::ConfidentialTransferAccount>()?;
+
+        let destination_state = self
+            .get_account_info(destination_token_account)
+            .await
+            .unwrap();
+        let destination_extension = destination_state
+            .get_extension::<confidential_transfer::ConfidentialTransferAccount>(
+        )?;
+
+        let mint_state = self.get_mint_info().await.unwrap();
+        let transfer_fee_config = mint_state
+            .get_extension::<transfer_fee::TransferFeeConfig>()
+            .unwrap();
+
+        // Zkp generator requires transfer fee parameters as input
+        let fee_parameters =
+            if u64::from(transfer_fee_config.newer_transfer_fee.epoch) < epoch_info.epoch {
```

Comment:
It looks like you could use `transfer_fee_config.get_epoch_fee(epoch_info.epoch)` to obtain the appropriate `TransferFee`, which'll save a couple lines of code here

Reply:
Yep this is applied now, thanks!